src/components/table-hoc/TableWithFilter.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import {keys} from 'ts-transformer-keys/index';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {ReduxConnect} from '../../utils/ReduxUtils';
+import {FilterBoxConnected} from '../filterBox/FilterBoxConnected';
+import {IFilterState} from '../filterBox/FilterBoxReducers';
+import {ITableHOCOwnProps} from './TableHOC';
+
+export interface ITableWithFilterConfig {
+    matchFilter?: (filterValue: string, datum: any) => boolean;
+}
+
+export interface ITableWithFilterStateProps {
+    filter: string;
+}
+
+export interface ITableWithFilterDispatchProps {
+    onRender: () => void;
+}
+
+export interface ITableWithFilterProps extends Partial<ITableWithFilterStateProps>,
+    Partial<ITableWithFilterDispatchProps>,
+    ITableHOCOwnProps {}
+
+const TableWithFilterPropsToOmit = keys<ITableWithFilterStateProps>();
+
+const defaultMatchFilter = (filter: string, datum: any) => JSON.stringify(_.values(datum).map((v: string) => v.toLowerCase())).indexOf(filter.toLowerCase()) !== -1;
+
+export const tableWithFilter = (config: ITableWithFilterConfig = {matchFilter: defaultMatchFilter}) => (Component: (React.ComponentClass<ITableWithFilterProps> | React.StatelessComponent<ITableWithFilterProps>)): React.ComponentClass<ITableWithFilterProps & React.HTMLAttributes<HTMLTableElement>> => {
+    const mapStateToProps = (state: IReactVaporState, ownProps: ITableWithFilterProps): ITableWithFilterStateProps | ITableHOCOwnProps => {

```

Comment:
This logic should eventually go in a cached selector since filtering can be expensive.